### PR TITLE
Fix _load_multiple_inputs in upath_io_manager

### PR DESCRIPTION
**Pull request by harrylojames**
## Summary & Motivation

Hi, hope this pull request is okay. When trying to load multipartitions paths[partition_key] of _load_multiple_inputs in upath_io_manager throws a key error. partition_key looks like 'a|1' whereas the keys of paths look like 'a/1'. Let me know if there is a more readable in built way to change strings between the two formats! 

`obj = self._load_partition_from_path(
                    context, paths[partition_key], backcompat_paths.get(partition_key)
                )`

## How I Tested These Changes

Added an additional test to test_upath_io_manager to ensure an upstream asset with multipartitons can be read in correctly.

### CODE REVIEW

**Comment by sryza** [comment]
@clairelin135 mind taking a look at this one? Do we have special handling for multi-partitioned assets in the default IO managers? Should we?

File: python_modules/dagster/dagster/_core/storage/upath_io_manager.py [hunk 1]
```diff
@@ -203,7 +203,7 @@ def _formatted_multipartitioned_path(partition_key: MultiPartitionKey) -> str:
 
         asset_path = self._get_path_without_extension(context)
         return {
-            partition: self._with_extension(
+            partition.replace("/", "|"): self._with_extension(
```
Comment (clairelin135) [inline python_modules/dagster/dagster/_core/storage/upath_io_manager.py:206]: Thoughts on instead refactoring `formatted_partition_keys` above to map the partition key to the formatted path?

Then, here we can just do 
```
{
    partition: self._with_extension(
        self.get_path_for_partition(context, asset_path, partition_path)
    )
    for partition, partition_path in formatted_partition_keys.items()
}
```
so we can isolate all multipartition specific logic to `_formatted_multipartitioned_path`

**Review by clairelin135 — COMMENTED** [review]
Thanks @harrylojames for catching this. Yep, looks like a bug when we try to load multiple multipartitioned inputs--we're mapping an incorrectly formatted partition key to the input object.

I added one small comment, but otherwise this looks good to me.

File: python_modules/dagster/dagster/_core/storage/upath_io_manager.py (hunk 1 above)
Comment (harrylojames) [reply python_modules/dagster/dagster/_core/storage/upath_io_manager.py]: Thank you for your comment. Let me know if the changes look okay

**Review by clairelin135 — APPROVED** [review]
This looks great, thank you!

**Pull request merged**